From 2bc84fa2a4db0efcdb78dfefc47d08cb7724dad8 Mon Sep 17 00:00:00 2001
From: David Daney <david.daney@cavium.com>
Date: Mon, 28 Jan 2013 14:33:16 -0800
Subject: [PATCH 008/122] MIPS: Allow r4k_switch.S to be included by
 octeon_switch.S

Source: Cavium Networks, Inc.
MR: 00000
Type: Integration
Disposition: Merged from Octeon Tree
ChangeID: 928182888f175535e8007e9803844067a107be0c
Description:

The follow-on patch includes r4k_switch.S from octeon_switch.S, use
the preprocessor symbol USE_ALTERNATE_RESUME_IMPL to disable the parts
of r4k_switch.S that are unwanted (the resume function).

Signed-off-by: David Daney <david.daney@cavium.com>
Signed-off-by: Corey Minyard <cminyard@mvista.com>
[Original patch taken from Cavium SDK 3.1.1 525]
Signed-off-by: Bin Jiang <bin.jiang@windriver.com>
---
 arch/mips/kernel/r4k_switch.S | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/mips/kernel/r4k_switch.S b/arch/mips/kernel/r4k_switch.S
index cc78dd9..2e4db00 100644
--- a/arch/mips/kernel/r4k_switch.S
+++ b/arch/mips/kernel/r4k_switch.S
@@ -28,6 +28,7 @@
  */
 #define ST_OFF (_THREAD_SIZE - 32 - PT_SIZE + PT_STATUS)
 
+#ifndef USE_ALTERNATE_RESUME_IMPL
 /*
  * FPU context is saved iff the process has used it's FPU in the current
  * time slice as indicated by _TIF_USEDFPU.  In any case, the CU1 bit for user
@@ -119,6 +120,8 @@
 	jr	ra
 	END(resume)
 
+#endif /* USE_ALTERNATE_RESUME_IMPL */
+
 /*
  * Save a thread's fp context.
  */
-- 
1.8.2.1

